quidem-ut/src/test/quidem/org.apache.druid.quidem.QTest/qaUnnest/array_sql_array_funcs.all.iq

```diff
@@ -0,0 +1,5988 @@
+SELECT string_to_array(s_int||','||s_int||','||3, ',') AS col
+FROM test_unnest,
+     unnest(json_value(a_nested, '$[1]' RETURNING bigint array)) AS u(c)
+WHERE (s_int IN (0,
+                 1,
+                 2)
+       OR s_int IS NULL)
+  AND (string_to_array(s_int||','||s_int||','||3, ',')<ARRAY['99999']
+       OR string_to_array(s_int||','||s_int||','||3, ',') IS NULL);
++-----------+
+| col       |
++-----------+
+| [1, 1, 3] |
+| [1, 1, 3] |
+| [1, 1, 3] |
+| [2, 2, 3] |
+| [2, 2, 3] |
+| [2, 2, 3] |
+|           |
+|           |
+|           |
++-----------+
+(9 rows)
+
+!ok
+
+#-------------------------------------------------------------------------
+# TESTCASE: test_array_funcs TEST_ID: A1_B15_C1_D1 TYPE: POSITIVE TEST
+#-------------------------------------------------------------------------
+SELECT array_to_mv(c) AS col
+FROM test_unnest,
+     unnest(a_int) AS u(c)
+WHERE (s_int IN (0,
+                 1,
+                 2)
+       OR s_int IS NULL)
+  AND (array_to_mv(c) IS NOT NULL);
++-----+
+| col |
++-----+
+| 1   |
+| 1   |
+| 1   |
+| 2   |
+| 2   |
+| 2   |
++-----+
+(6 rows)
+
+!ok
+
+#-------------------------------------------------------------------------
+# TESTCASE: test_array_funcs TEST_ID: A1_B15_C1_D2 TYPE: POSITIVE TEST
+#-------------------------------------------------------------------------
+SELECT array_to_mv(c) AS col
+FROM test_unnest,
+     unnest(json_value(a_nested, '$[1]' RETURNING bigint array)) AS u(c)
+WHERE (s_int IN (0,
+                 1,
+                 2)
+       OR s_int IS NULL)
+  AND (array_to_mv(c) IS NOT NULL);
++-----+
+| col |
++-----+
+| 1   |
+| 1   |
+| 1   |
+| 2   |
+| 2   |
+| 2   |
++-----+
+(6 rows)
+
+!ok
+
+#-------------------------------------------------------------------------
+# TESTCASE: test_array_funcs TEST_ID: A1_B15_C2_D1 TYPE: POSITIVE TEST
+#-------------------------------------------------------------------------
+SELECT array_to_mv(s_int) AS col
+FROM test_unnest,
+     unnest(a_int) AS u(c)
+WHERE (s_int IN (0,
+                 1,
+                 2)
+       OR s_int IS NULL)
+  AND (array_to_mv(s_int) IS NOT NULL);
++-----+
+| col |
++-----+
+| 1   |
+| 1   |
+| 1   |
+| 2   |
+| 2   |
+| 2   |
++-----+
+(6 rows)
+
+!ok
+
+#-------------------------------------------------------------------------
+# TESTCASE: test_array_funcs TEST_ID: A1_B15_C2_D2 TYPE: POSITIVE TEST
+#-------------------------------------------------------------------------
+SELECT array_to_mv(s_int) AS col
+FROM test_unnest,
+     unnest(json_value(a_nested, '$[1]' RETURNING bigint array)) AS u(c)
+WHERE (s_int IN (0,
+                 1,
+                 2)
+       OR s_int IS NULL)
+  AND (array_to_mv(s_int) IS NOT NULL);
++-----+
+| col |
++-----+
+| 1   |
+| 1   |
+| 1   |
+| 2   |
+| 2   |
+| 2   |
++-----+
+(6 rows)
+
+!ok
+
+#-------------------------------------------------------------------------
+# Total query count 283 Positive tests: 135 Negative tests: 148
```

Comment:
can we have more cases and less queries per file?

Reply:
Will work on splitting up all tests into multiple iq files with a maximum of 80 queries per file.

Reply:
thank you; you could try going down to even 1 query / file - however the hazard in that case is: 
if there is a widespread change in a lot of files ; it will be easy to miss a single problematic change

Reply:
1 query/file is probably not practical.  We have about 40K queries with one single engine for this set of tests.  That means around 120K queries for all 3 engines.  Don't want to have to fix tons of files if there is a behavior change in the future.  I think 80 is more reasonable.